src/BotPlutusInterface/Contract.hs

```diff
@@ -181,27 +184,17 @@ writeBalancedTx _ (Left _) = error "Cannot handle cardano api tx"
 writeBalancedTx contractEnv (Right tx) = do
   createDirectoryIfMissing @w False (Text.unpack contractEnv.cePABConfig.pcScriptFileDir)
 
-  fileWriteRes <-
-    Files.writeAll @w contractEnv.cePABConfig tx
+  eitherT (pure . WriteBalancedTxFailed . OtherError) (pure . WriteBalancedTxSuccess . Right) $ do
+    void $ firstEitherT (Text.pack . show) $ newEitherT $ Files.writeAll @w contractEnv.cePABConfig tx
```

Comment:
Theres a lot going on here, but I can't think of a way to simplify when using EitherT :(